### PR TITLE
sorting, ui enhancements

### PR DESCRIPTION
**Pull request by ngour5**
- Implemented local sorting since we don't need to bubble event upwards.
- Removed back button and instead, added "Top Results" option in left nav.
- Did some UI enhancements
- Stripped HTML tags in table (only for demo) since lightning-datatable doesn't support rich text

### CODE REVIEW

File: src/entitySearchResultsList.js [hunk 1]
```diff
@@ -60,6 +60,9 @@ export default class EntitySearchResultsList extends LightningElement {
                     if(originalValue instanceof Object){
                         key = key.replaceAll("\.", "-");
                     }
+                    
+                    val = val.replace("<br>", " ");
+                    val = val.replace(/<\/?[^>]+(>|$)/g, "");
```
Comment (ngour5) [inline src/entitySearchResultsList.js:65]: Removing html tags only for demo

File: src/entitySearchResultsList.js [hunk 2]
```diff
@@ -85,6 +88,7 @@ export default class EntitySearchResultsList extends LightningElement {
                 copyField.fieldName = copyField.fieldName.split(".")[0];
             }
 
+            copyField.sortable = true;
```
Comment (ngour5) [inline src/entitySearchResultsList.js:91]: Setting all fields to be sortable

File: src/entitySearchResultsList.js [hunk 3]
```diff
@@ -98,7 +102,27 @@ export default class EntitySearchResultsList extends LightningElement {
...
 
-    handleSort(evt){
-        this.dispatchEvent(new CustomEvent('sort', {detail: {name: "juan"}}));
+    handleSort({ detail }) {
+        this.sortedBy = detail.fieldName;
+        this.sortDirection = detail.sortDirection;
+        this.performSort(this.sortedBy, this.sortDirection);
+    }
+
+    performSort(sortColumn, sortDirection) {
```
Comment (ngour5) [inline src/entitySearchResultsList.js:111]: sorting locally.

File: src/entitySearchResultsList.js [hunk 4]
```diff
@@ -98,7 +102,27 @@ export default class EntitySearchResultsList extends LightningElement {
         this.dispatchEvent(new CustomEvent('loadmore', {detail: {name: "juan"}}));
     }
 
-    handleSort(evt){
-        this.dispatchEvent(new CustomEvent('sort', {detail: {name: "juan"}}));
+    handleSort({ detail }) {
+        this.sortedBy = detail.fieldName;
+        this.sortDirection = detail.sortDirection;
```
Comment (ngour5) [inline src/entitySearchResultsList.js:107]: Normally, setting `this.sortedBy` and `this.sortDirection` on `lightning:datatable` should be enough to sort the table. I'm not sure why its not working, so we can just sort locally.

File: src/searchableEntitiesList.html [hunk 5]
```diff
@@ -1,23 +1,20 @@
...
     </div>
 
-    <span class="slds-p-bottom_medium">Total entities: {totalCount}</span>
-
     <lightning-spinner alternative-text="Loading" size="large" if:true={loadingEntities}></lightning-spinner>
 
     <div if:false={loadingEntities} class="scrollable">
-      <lightning-vertical-navigation onselect={handleSelect} class="navigation">
+      <lightning-vertical-navigation onselect={handleSelect} class="navigation" selected-item="top_results">
+        <lightning-vertical-navigation-item name="top_results" label="Top Results"></lightning-vertical-navigation-item>
```
Comment (ngour5) [inline src/searchableEntitiesList.html:13]: Added "Top Results" as an option in left nav

File: src/searchPage.html [hunk 6]
```diff
@@ -1,12 +1,11 @@
...
   <div onkeyup={handleKeyUp} class="search-bar-container">
-    <lightning-input name="enter-search" label="Type your search query here" type="search"
-      class="search-bar slds-p-bottom_medium" onchange={updateSearchTerm}></lightning-input>
+    <legend class="page-header">Search</legend>
+    <lightning-input name="enter-search" variant="label-hidden" label="Label" placeholder="Type your search query here"
+      type="search" class="search-bar slds-p-bottom_medium" onchange={updateSearchTerm}></lightning-input>
   </div>
   <div class="slds-grid slds-gutters">
-    <div class="slds-col slds-size_1-of-4">
-      <lightning-button if:true={isScopedSearch} variant="base" label="Back" title="Return to Top Results"
```
Comment (ngour5) [inline src/searchPage.html:8]: Removed back button. I think we can just keep Top Results to go back to it

File: src/entitySearchResultsList.js (hunk 4 above)
Comment (juanka588) [reply src/entitySearchResultsList.js:107]: Thanks

File: src/entitySearchResultsList.js (hunk 3 above)
Comment (juanka588) [reply src/entitySearchResultsList.js:111]: yeah I wanted to bubble up the event in order to send a new ask request with the sorting parameters, something to explore after the hack I suppose

**Review by juanka588 — APPROVED** [review]

File: src/entitySearchResultsList.js (hunk 1 above)
Comment (juanka588) [reply src/entitySearchResultsList.js:65]: In real life I suppose this is why knowledge behaves differently than other entities

**Pull request merged**